cmd/consoleplugin/main.go

```diff
@@ -1,5 +1,125 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"syscall"
+)
+
+var setupLog = ctrl.Log.WithName("setup")
+
 func main() {
 
+	opts := zap.Options{
+		Development: true,
+		Encoder: zapcore.NewJSONEncoder(zapcore.EncoderConfig{
+			TimeKey:        "ts",
+			LevelKey:       "level",
+			NameKey:        "logger",
+			CallerKey:      "caller",
+			FunctionKey:    zapcore.OmitKey,
+			MessageKey:     "msg",
+			StacktraceKey:  "stacktrace",
+			LineEnding:     zapcore.DefaultLineEnding,
+			EncodeLevel:    zapcore.LowercaseLevelEncoder,
+			EncodeTime:     zapcore.ISO8601TimeEncoder,
+			EncodeDuration: zapcore.SecondsDurationEncoder,
+			EncodeCaller:   zapcore.ShortCallerEncoder,
+		}),
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	// also set the client-go logger so we get the same JSON output
+	klogv2.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	// see https://github.com/kubernetes/klog#coexisting-with-klogv2
+	// BEGIN : hack to redirect klogv1 calls to klog v2
+	// Tell klog NOT to log into STDERR. Otherwise, we risk
+	// certain kinds of API errors getting logged into a directory not
+	// available in a `FROM scratch` Docker container, causing us to abort
+	var klogv1Flags flag.FlagSet
+	klogv1.InitFlags(&klogv1Flags)
+	if err := klogv1Flags.Set("logtostderr", "false"); err != nil { // By default klog v1 logs to stderr, switch that off
+		setupLog.Error(err, "")
+		os.Exit(1)
+	}
+	if err := klogv1Flags.Set("stderrthreshold", "FATAL"); err != nil { // stderrthreshold defaults to ERROR, so we don't get anything in stderr
+		setupLog.Error(err, "")
+		os.Exit(1)
+	}
+	klogv1.SetOutputBySeverity("INFO", klog.Writer{}) // tell klog v1 to use the custom writer
+	// END : hack to redirect klogv1 calls to klog v2
+
+	setupLog.Info("Configuring web console plugin server ...")
+
+	runtimeScheme := runtime.NewScheme()
+	cfg, err := config.GetConfig()
+	if err != nil {
+		setupLog.Error(err, "getting config failed")
+		os.Exit(1)
+	}
+	err = userv1.Install(runtimeScheme)
+	if err != nil {
+		setupLog.Error(err, "adding user to scheme failed")
+		os.Exit(1)
+	}
+
+	_, err = client.New(cfg, client.Options{
+		Scheme: runtimeScheme,
+	})
+	if err != nil {
+		setupLog.Error(err, "creating a new client failed")
+		os.Exit(1)
+	}
+
+	mux := http.NewServeMux()
+
+	s := scriptserver.NewScriptServer()
+
+	mux.HandleFunc("/", s.HandleScriptRequest)
+	mux.HandleFunc("/api/v1/status", healthcheck.HandleHealthCheck)
+
+	consolePluginServer := &http.Server{ //nolint:gosec
+		Addr:    ":8080",
+		Handler: mux,
+	}
+
+	setupLog.Info("Web Console Plugin server configured.")
+
+	go func() {
+		setupLog.Info("Listening...")
+
+		if err := consolePluginServer.ListenAndServeTLS("/etc/webhook/certs/"+cert.ServerCert, "/etc/webhook/certs/"+cert.ServerKey); err != nil {
```

Comment:
I think we shouldn't start a TLS service. I guess you copied it from the web hook service but it's different. Admition web hook are special services in Kube. Here we need a simple http service.
Unless web console does require an internal service running with TLS... Which I doubt.